docs/proposals/signal-api.md

```diff
@@ -0,0 +1,123 @@
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Signal API
+
+This document describes signal API effort in Kubernetes.
+
+## Goal
+
+The goal is to provide a possibility to send signals to container PID 1 processes. At the beginning, we
+are going to support:
+- SIGHUP
+- SIGKILL
+- SIGTERM
+Once me make agreement on the API part, the document will be extended with the implementation details.
+
+#### Containers and pods
+
+Despite the fact that Docker supports shared PID namespaces, Kubernetes doesn't, which means that signals can be
+sent only to concrete containers, not pods.
+
+## Main use cases
+
+#### Stopping applications
+
+Currently there is no way to tell applications in containers in pods to stop without deleting a pod. Assuming that
+a pod has *Always* or *OnFailure* restartPolicy, we can make a possibility of manual restart for pods by sending the
+SIGTERM signal.
+
+#### Force stopping applications
+
+Similarly, we can make a possibility of manual force restart by sendinf the SIGKILL signal.
+
+#### Re-reading configuration
+
+Many applications follow the convention to re-read configuration files when handling the SIGHUP signal. Ability
+to send SIGHUP signal will allow us to reconfigure pods without updating their definition or removing them. Also,
+we can consider using this API in the watch mechanism in kubelet to reconfigure a pod automatically on the ConfigMap
+change.
+
+## Proposed API
+
+Signal API will be a part of pod API. There will be no separate API group for that.
+
+#### Schema
+
+The proposed schema is as follow.
+
+There will be subresources of `pod` called `BumpREST` and `StopREST`, defined like:
+
+```go
+type BumpREST struct {
+	Store       *registry.Store
+	KubeletConn client.ConnectionInfoGetter
+}
+
+type StopREST struct {
+	Store       *registry.Store
+	KubeletConn client.ConnectionInfoGetter
+}
+```
+
+And there will be types for options for each subresource:
+
+```go
+type PodBumpOptions struct {
+	unversioned TypeMeta `json:",inline"`
+	Container string `json:"container,omitempty" protobuf:"bytes,1,opt,name=container"`
+}
+
+type PodStopOptions struct {
+	unversioned TypeMeta `json:",inline"`
+	Force bool `json:"force,omitempty" protobuf:"varint,1,opt,name=force"`
+	Container string `json:"container,omitempty" protobuf:"bytes,2,opt,name=container"`
+}
+```
+
+#### Endpoints
+
+All endpoints are GET endpoints, rooted at /api/v1/.
```

Comment:
this seems to break GET being a safe action.

this needs to be a PUT/POST at minimum.